docs/treeview_performance.md

```diff
@@ -0,0 +1,136 @@
+
+-   Initial render time
+-   Time when expanding a Node
+-   Time when selecting a Node
+-   Time to start dragging an item
+
+> The numbers are considered using Chrome's DevTools Performance tool, considering the long task which includes all React operations from event to repaint and DOM update.
+
+| Initial rendering | Expand a node  | Select a node  |
+| ----------------- | -------------- | -------------- |
+| 1162ms / 5 ops    | 1221ms / 8 ops | 1108ms / 8 ops |
+
+The above results bring some interesting results, it also means that the more data will be slower, there is room for improvement and avoid some use cases. Let's take a closer look at the scenarios and possible problems.
+
+#### Initial rendering
+
+<img alt="Tracing initial rendering" src="https://user-images.githubusercontent.com/13750819/140421402-5c6ad4ed-f640-4c34-af7c-4ebca89e0509.png" style="border-radius: 5px;" />
+
+The render phase has a considerably noticeable time which is slow for the user considering the RAIL model. This is very dependent on the number of components being rendered in a list, the tracing above shows well that the number of components being rendered but the main thread being occupied to render the complete list.
+
+This initial rendering time can increase if a list of expanded Nodes is defined, this example is rendering the first level of TreeView with 550 Nodes, in the number of components, this is higher because only one Node can render more components per Node, consider 10 components by Node.
+
+Some alternatives we can adopt here to solve this is virtualization as recommended by the React.js team which will only render a limited amount of Nodes according to the viewport.
+
+#### Expand a node
+
+<img alt="Tracing expand a node" src="https://user-images.githubusercontent.com/13750819/140421389-a2098c37-5e5d-42c0-a9a9-4d54839b67ed.png" style="border-radius: 5px;" />
+
+The API decision to avoid tree traversal to change a property on each Node that would symbolize whether the Node is expanded or not also has the same impact as causing a render on all other Nodes but avoids the cost of traversing the tree.
+
+Since all Nodes consume the same context to know if the Node is expanded any new items added to the state will cause it to re-render.
+
+Ideally, just cause a rendering on the Node that is being expanded, we have some alternatives to avoid that like treating the `expandedKeys` only as initial and using local state on the Node to expand, we can have some drastic performance gains.
+
+This is a common problem for context usage that all-consuming components will cause re-rendering, there is a proposal for this in [React RFC 119 (Context selectors)](https://github.com/reactjs/rfcs/pull/119) that would render the component only if the data it is consuming has changed, a [PR from the React team](https://github.com/facebook/react/pull/20646) was sent some time ago as an experiment but still open.
```

Comment:
There are some solutions in userspace that try to implement the idea of Context selectors to not render all components just if the data it's consuming has changed. I'm working on an implementation for this too.

Reply:
Well I used a userspace ([use-context-selector](https://github.com/dai-shi/use-context-selector)) implementation that already implements this idea but I haven't had great performance results, I can see fewer components not being rendered but I can't see a drop in render time.